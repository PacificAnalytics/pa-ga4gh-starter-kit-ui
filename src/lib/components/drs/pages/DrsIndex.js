import React from 'react';
import { 
  Typography, 
  Container, 
  Table,
  TableContainer, 
  TableHead, 
  TableBody, 
  TableRow, 
  TableCell, 
  Button, 
  Grid, 
  IconButton
} from '@material-ui/core';
import {Link} from "react-router-dom";
import { scrollToTop } from '../../../functions/common';
import EditIcon from '@material-ui/icons/Edit';

  /* Render index table rows populated with data */
const DrsIndexRows = (props) => {
  const drsObjectsList = props.drsObjectsList;
  if (!drsObjectsList){
    return null;
  }
  else {
    const rows = drsObjectsList.map((drsObject) =>
    <TableRow key={drsObject.id}>
      <TableCell align="left">
        <Typography>{drsObject.id}</Typography>
      </TableCell>
      <TableCell align="left">
        <Typography>{drsObject.name}</Typography>
      </TableCell>
      <TableCell align="right">
        <Button
          aria-label={`view-details-button`}
          variant="contained"
          color='default'
          component={Link}
          to={`/drs/${drsObject.id}`}
          onClick={scrollToTop}
        >
          <Typography variant="button">View Details</Typography>
        </Button>
      </TableCell>
      <TableCell align="center">
        <IconButton 
          aria-label={`edit-button`}
          variant="contained" 
          color='primary' 
          component={Link} 
          to={`/drs/${drsObject.id}/edit`}
          onClick={scrollToTop}
        >
          <EditIcon/>
        </IconButton>
      </TableCell>
    </TableRow>
    );
    return (
      <TableBody>{rows}</TableBody>
    );
  }
}

const DrsIndex = (props) => {
  /* Render DrsIndex page */
  return (
    <div align="center">
      <meta
          name="viewport"
          content="minimum-scale=1, initial-scale=1, width=device-width"
      />
      <Container maxWidth="lg" >
        <Grid container justify='space-between' alignItems='center'>
          <Grid item xs={2} align='left'>
            <Button 
              variant='contained' 
              component={Link} 
              to='/' 
              color='primary' 
              size='large'
              onClick={scrollToTop}
            >
              <Typography variant='button'>Home</Typography>
            </Button>
          </Grid>
          <Grid item xs={8}>
            <Typography variant="h2" gutterBottom>Welcome to DRS Starter Kit</Typography>
          </Grid>
          <Grid item xs={2} align='right'>
            <Button 
              variant='contained' 
              component={Link} 
              to='/drs/new' 
              color='primary' 
              size='large'
              onClick={scrollToTop}
            >
              <Typography variant='button'>New DRS Object</Typography>
            </Button>
          </Grid>
        </Grid>
      </Container>
      <Container maxWidth="lg">
        <TableContainer>
          <Table>
            <TableHead>
              <TableRow>
                <TableCell align="left">
                  <Typography variant="h5">DRS Object ID</Typography>
                </TableCell>
                <TableCell align="left">
                  <Typography variant="h5">Name</Typography>
                </TableCell>
                <TableCell align="center">
                  <Typography variant="h5">View Details</Typography>
                </TableCell>
                <TableCell align="center">
                  <Typography variant="h5">Edit</Typography>
                </TableCell>
              </TableRow>
            </TableHead>
            <DrsIndexRows drsObjectsList={props.drsObjectsList}/>
          </Table>
        </TableContainer>
      </Container>
    </div>
  );
}

export default DrsIndex;