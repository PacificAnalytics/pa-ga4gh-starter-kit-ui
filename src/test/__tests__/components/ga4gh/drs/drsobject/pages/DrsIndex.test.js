import React from 'react';
import renderer from 'react-test-renderer';
import {render, screen, getByRole, queryByRole, getByLabelText, waitFor, logRoles, getByText} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import userEvent from '@testing-library/user-event';
import {
    mockDrsObjectsList
} from '../../../../../../resources/MockData';
import DrsIndex from '../../../../../lib/components/drs/pages/DrsIndex';
import { MemoryRouter } from 'react-router';

test('Render <DrsIndex /> component with data', () => {
    let drsObjectsList = mockDrsObjectsList
    let {container} = render(
        <MemoryRouter>
            <DrsIndex drsObjectsList={drsObjectsList} />
        </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByRole('heading', {level: 2})).toHaveTextContent('Welcome to DRS Starter Kit');
    expect(screen.getByRole('heading', {name: 'DRS Object ID'})).toBeInTheDocument();
    expect(screen.getByRole('heading', {name: 'Name'})).toBeInTheDocument();
    expect(screen.getByRole('heading', {name: 'View Details'})).toBeInTheDocument();
    expect(screen.getByRole('heading', {name: 'Edit'})).toBeInTheDocument();

    drsObjectsList.forEach((drsObject) => {
        let indexRow = screen.getByRole('row', {name: `${drsObject.id} ${drsObject.name} View Details`})
        expect(getByText(indexRow, `${drsObject.id}`)).toBeInTheDocument();
        expect(getByText(indexRow, `${drsObject.name}`)).toBeInTheDocument();
        expect(getByLabelText(indexRow, 'view-details-button')).toBeInTheDocument();
        expect(getByLabelText(indexRow, 'edit-button')).toBeInTheDocument();
    });
});

test('Render <DrsIndex /> component without data', () => {
    let {container} = render(
        <MemoryRouter>
            <DrsIndex drsObjectsList={null} />
        </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(screen.getByRole('heading', {level: 2})).toHaveTextContent('Welcome to DRS Starter Kit');
    expect(screen.getByText('DRS Object ID')).toBeInTheDocument();
    expect(screen.getByText('Name')).toBeInTheDocument();
    expect(screen.getByText('View Details')).toBeInTheDocument();
    expect(screen.getByText('Edit')).toBeInTheDocument();
    expect(screen.queryByLabelText('view-details-button')).not.toBeInTheDocument();
    expect(screen.queryByLabelText('edit-button')).not.toBeInTheDocument();
});