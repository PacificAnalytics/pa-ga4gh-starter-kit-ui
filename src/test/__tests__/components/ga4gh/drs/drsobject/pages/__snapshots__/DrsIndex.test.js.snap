// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render <DrsIndex /> component with data 1`] = `
<div
  align="center"
>
  <meta
    content="minimum-scale=1, initial-scale=1, width=device-width"
    name="viewport"
  />
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
    >
      <div
        align="left"
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
          href="/"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiTypography-root MuiTypography-button"
            >
              Home
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
        >
          Welcome to DRS Starter Kit
        </h2>
      </div>
      <div
        align="right"
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
          href="/drs/new"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiTypography-root MuiTypography-button"
            >
              New DRS Object
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiTableContainer-root"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                DRS Object ID
              </h5>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                Name
              </h5>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                View Details
              </h5>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                Edit
              </h5>
            </th>
          </tr>
        </thead>
        <tbody
          class="MuiTableBody-root"
        >
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                b8cd0667-2c33-4c9f-967b-161b905932c9
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                phenopackets.test.dataset
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              <a
                aria-disabled="false"
                aria-label="view-details-button"
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                href="/drs/b8cd0667-2c33-4c9f-967b-161b905932c9"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiTypography-root MuiTypography-button"
                  >
                    View Details
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              <a
                aria-disabled="false"
                aria-label="edit-button"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                href="/drs/b8cd0667-2c33-4c9f-967b-161b905932c9/edit"
                role="button"
                tabindex="0"
                variant="contained"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                a1dd4ae2-8d26-43b0-a199-342b64c7dff6
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                phenopackets.cao.family
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              <a
                aria-disabled="false"
                aria-label="view-details-button"
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                href="/drs/a1dd4ae2-8d26-43b0-a199-342b64c7dff6"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiTypography-root MuiTypography-button"
                  >
                    View Details
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              <a
                aria-disabled="false"
                aria-label="edit-button"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                href="/drs/a1dd4ae2-8d26-43b0-a199-342b64c7dff6/edit"
                role="button"
                tabindex="0"
                variant="contained"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                1a570e4e-2489-4218-9333-f65549495872
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                phenopackets.cao.1
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              <a
                aria-disabled="false"
                aria-label="view-details-button"
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                href="/drs/1a570e4e-2489-4218-9333-f65549495872"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiTypography-root MuiTypography-button"
                  >
                    View Details
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              <a
                aria-disabled="false"
                aria-label="edit-button"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                href="/drs/1a570e4e-2489-4218-9333-f65549495872/edit"
                role="button"
                tabindex="0"
                variant="contained"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                4d83ba3f-a476-4c7c-868f-3d1fcf77fe29
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                phenopackets.cao.2
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              <a
                aria-disabled="false"
                aria-label="view-details-button"
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                href="/drs/4d83ba3f-a476-4c7c-868f-3d1fcf77fe29"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiTypography-root MuiTypography-button"
                  >
                    View Details
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              <a
                aria-disabled="false"
                aria-label="edit-button"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                href="/drs/4d83ba3f-a476-4c7c-868f-3d1fcf77fe29/edit"
                role="button"
                tabindex="0"
                variant="contained"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                924901d5-6d31-4c33-b443-7931eadfac4b
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                phenopackets.cao.3
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              <a
                aria-disabled="false"
                aria-label="view-details-button"
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                href="/drs/924901d5-6d31-4c33-b443-7931eadfac4b"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiTypography-root MuiTypography-button"
                  >
                    View Details
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              <a
                aria-disabled="false"
                aria-label="edit-button"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                href="/drs/924901d5-6d31-4c33-b443-7931eadfac4b/edit"
                role="button"
                tabindex="0"
                variant="contained"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </td>
          </tr>
          <tr
            class="MuiTableRow-root"
          >
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                0f8abce3-e161-4bdf-981f-86257d505d69
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft"
            >
              <p
                class="MuiTypography-root MuiTypography-body1"
              >
                phenopackets.cao.4
              </p>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
            >
              <a
                aria-disabled="false"
                aria-label="view-details-button"
                class="MuiButtonBase-root MuiButton-root MuiButton-contained"
                href="/drs/0f8abce3-e161-4bdf-981f-86257d505d69"
                role="button"
                tabindex="0"
              >
                <span
                  class="MuiButton-label"
                >
                  <span
                    class="MuiTypography-root MuiTypography-button"
                  >
                    View Details
                  </span>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </td>
            <td
              class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignCenter"
            >
              <a
                aria-disabled="false"
                aria-label="edit-button"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorPrimary"
                href="/drs/0f8abce3-e161-4bdf-981f-86257d505d69/edit"
                role="button"
                tabindex="0"
                variant="contained"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
                    />
                  </svg>
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </a>
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`Render <DrsIndex /> component without data 1`] = `
<div
  align="center"
>
  <meta
    content="minimum-scale=1, initial-scale=1, width=device-width"
    name="viewport"
  />
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-space-between"
    >
      <div
        align="left"
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
          href="/"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiTypography-root MuiTypography-button"
            >
              Home
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-8"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom"
        >
          Welcome to DRS Starter Kit
        </h2>
      </div>
      <div
        align="right"
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2"
      >
        <a
          aria-disabled="false"
          class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-containedSizeLarge MuiButton-sizeLarge"
          href="/drs/new"
          role="button"
          tabindex="0"
        >
          <span
            class="MuiButton-label"
          >
            <span
              class="MuiTypography-root MuiTypography-button"
            >
              New DRS Object
            </span>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthLg"
  >
    <div
      class="MuiTableContainer-root"
    >
      <table
        class="MuiTable-root"
      >
        <thead
          class="MuiTableHead-root"
        >
          <tr
            class="MuiTableRow-root MuiTableRow-head"
          >
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                DRS Object ID
              </h5>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft"
              scope="col"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                Name
              </h5>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                View Details
              </h5>
            </th>
            <th
              class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignCenter"
              scope="col"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5"
              >
                Edit
              </h5>
            </th>
          </tr>
        </thead>
      </table>
    </div>
  </div>
</div>
`;
